packages/plugins/quorum/src/deployer.ts

```diff
@@ -0,0 +1,92 @@
+import { __ } from 'embark-i18n';
+import Web3 from 'web3';
+import { Embark, EmbarkEvents } from 'embark-core';
+import { Logger } from 'embark-logger';
+const embarkJsUtils = require('embarkjs').Utils;
+
+export default class QuorumDeployer {
+  private logger: Logger;
+  private events: EmbarkEvents;
+  private _web3: Web3 | null = null;
+  constructor(private embark: Embark) {
+    this.logger = embark.logger;
+    this.events = embark.events;
+  }
+
+  get web3() {
+    return (async () => {
+      if (!this._web3) {
+        const provider = await this.events.request2("blockchain:client:provider", "ethereum");
+        this._web3 = new Web3(provider);
+      }
+      return this._web3;
+    })();
+  }
+
+  public registerDeployer() {
+    this.events.request("deployment:deployer:register", "ethereum", this.deployer.bind(this));
+    this.embark.registerActionForEvent("deployment:contract:shouldDeploy", { priority: 40 }, this.skipBytecodeCheck.bind(this));
+  }
+
+  skipBytecodeCheck(params, cb) {
+    const skipBytecodeCheck = this.embark.config.contractsConfig?.contracts[params.contract.className]?.skipBytecodeCheck ?? false;
+    if (skipBytecodeCheck) {
+      this.logger.warn(__("WARNING: Skipping bytecode check for %s deployment. Performing an embark reset may cause the contract to be re-deployed to the current Quorum node regardless if it was already deployed on another node.", params.contract.className));
+    }
+    params.skipBytecodeCheck = skipBytecodeCheck;
+    cb(null, params);
+  }
+
+  async deployer(contract, done) {
+    try {
+      const web3 = await this.web3;
+      const [account] = await web3.eth.getAccounts();
+      const contractObj = new web3.eth.Contract(contract.abiDefinition, contract.address);
+      // @ts-ignore
+      contractObj._jsonInterface.find(obj => {
+        if (obj.type === 'constructor') {
+          if (!obj.name) {
+            // Add constructor as the name of the method, because if there is an error, it prints `undefined` otherwise
+            obj.name = 'constructor';
+          }
+          return true;
+        }
+        return false;
+      });
+      const code = contract.code.substring(0, 2) === '0x' ? contract.code : "0x" + contract.code;
+      const contractObject = contractObj.deploy({ arguments: (contract.args || []), data: code });
+      if (contract.gas === 'auto' || !contract.gas) {
+        const gasValue = await contractObject.estimateGas({ value: 0, from: account });
+        const increase_per = 1 + (Math.random() / 10.0);
```

Comment:
Any reason this is snake_case'd?

Reply:
This was copied from https://github.com/embarklabs/embark/blob/ead92bd90f51b2796aee8a3feb88929feb64976b/packages/plugins/ethereum-blockchain-client/src/index.js#L63-L108, with some small additions for `privateFor`, `privateFrom`, and `bytecodeWithInitParam`.

Do you think we should update the snake case?

Reply:
I ended up removing this custom deployer function as per @jrainville 's suggestion in https://github.com/embarklabs/embark/pull/2299/files/045fddcd34112716ce37733f479a06e540ce982c#r386406028.